hangman/lib/game/Game.ex

```diff
@@ -0,0 +1,145 @@
+      2
+      iex> Hangman.Game.count(["a","p","p","l","e"])
+      0
+  """
+  def count(letters), do: Enum.reduce(letters,0,&valid?/2)
+
+  @doc """
+  update game's record by user's input
+  if :good_guess or :already_used , truns_left unchanging
+  ##Example
+      iex> Hangman.Game.update_record(["a","p","p","l","e"],%Hangman.Game.Game_record{letters: ["_","_","_","_","_"]},"p")
+      %Hangman.Game.Game_record{game_state: :good_guess, turns_left: 7,letters: ["_","p","p","_","_"], used: ["p"], last_guess: "p"}
+
+      iex> Hangman.Game.update_record(["a","p","p","l","e"],%Hangman.Game.Game_record{letters: ["_","_","_","_","_"]},"x")
+      %Hangman.Game.Game_record{game_state: :bad_guess, turns_left: 6,letters: ["_","_","_","_","_"], used: ["x"], last_guess: "x"}
+  """
+  def update_record(word,%Hangman.Game.Game_record{}=record,letter) do
+    cond do 
+      record.game_state == :won -> IO.puts "move is invalid after won"; record
+      record.game_state == :lost -> IO.puts "move is invalid after lost"; record
+      true ->
+      letters = record.letters
+      already_used = letter in record.used
+      record = %{record| letters: change_letters(word,record.letters,letter),
+		         last_guess: letter,
+		         used: [letter|record.used] |> Enum.sort
+		}
+      ret = count(record.letters)
+      bef = count(letters)
+      cond do
+        ret==0 -> %{record| game_state: :won}
+        record.turns_left==1 -> %{record| game_state: :lost,
+					          turns_left: record.turns_left-1,
+					          letters: word};
+        ret < bef -> %{record| game_state: :good_guess} 
+        already_used ->  %{record| game_state: :already_used}
+        true -> %{record| game_state: :bad_guess,
+		            turns_left: record.turns_left-1 }
+      end
+    end
+  end
+
+  # "used to receive and send message, like a simple server to maintain state"  
+  def play(word, record) do
+    receive do
+      {sender,:move,letter} ->
+	record=update_record(word,record,letter)
+	send sender, {self(),record |>Map.from_struct}
+	play(word,record)
+      {sender, :get_information} ->
+	send sender, record |> Map.from_struct
+	play(word,record)
+    end
+  end
+
+  def receive_msg() do
+    receive do
+      msg ->msg
+    end
+  end
+
+  @doc """
+  if string is empty return "cat", else return the string
+  ##Example
+     iex> Hangman.Game.deal_empty([])
+     "cat"
+     iex> Hangman.Game.deal_empty("dog")
+     "dog"
+     iex> Hangman.Game.deal_empty("swan")
+     "swan"
+
+  """
+  def deal_empty([]), do: "cat"
+  def deal_empty(str),do: str
+
+  # initialize a new game with a certain word
+  def init_game(word) do
+    word =
+      word|>
+      String.codepoints|>
+      Enum.filter(&(&1!="\r"))|> #avoid "\r"
+      deal_empty   #if word is empty ,return "cat"
+    #IO.inspect word 
+    initword = word |> Enum.reduce([],fn(_,acc)->["_"|acc] end)
+    spawn(Hangman.Game,:play,[word,%Game_record{letters: initword}])
```

Comment:
## +3 

You surprised me—you are the only person so far to write the game in a separate process.

In this case, I don't think that's actually necessary, but it is still a cool thing to have done.